infra/Pos/DHT/Real/Real.hs

```diff
@@ -72,77 +48,51 @@ stopDHTInstance KademliaDHTInstance {..} = liftIO $ K.close kdiHandle
 
 -- | Start 'KademliaDHTInstance' with 'KademliaParams'.
 startDHTInstance
-    :: ( MonadIO m
-       , MonadCatch m
-       , WithLogger m
-       , Bi DHTData
+    :: ( Bi DHTData
        , Bi DHTKey
        )
-    => KademliaParams
+    => Trace IO (Severity, Text)
+    -> KademliaParams
     -> NetworkAddress -- ^ Default NetworkAddress to bind.
-    -> m KademliaDHTInstance
-startDHTInstance kconf@KademliaParams {..} defaultBind = do
+    -> IO KademliaDHTInstance
+startDHTInstance logTrace kconf@KademliaParams {..} defaultBind = do
     let bindAddr = first B8.unpack (fromMaybe defaultBind kpNetworkAddress)
         extAddr  = maybe bindAddr (first B8.unpack) kpExternalAddress
-    logInfo "Generating dht key.."
+    traceWith logTrace (Info, "Generating dht key..")
     kdiKey <- maybe randomDHTKey pure kpKey
-    logInfoS $ sformat ("Generated dht key "%build) kdiKey
+    traceWith logTrace (Info, sformat ("Generated dht key "%build) kdiKey)
     kdiDumpPath <- case kpDumpFile of
         Nothing -> pure Nothing
         Just fp -> do
-            exists <- liftIO (doesFileExist fp)
+            exists <- doesFileExist fp
             pure $ if exists then Just fp else Nothing
     kdiHandle <- case kdiDumpPath of
         Just dumpFile -> do
-            logInfo "Restoring DHT Instance from snapshot"
+            traceWith logTrace (Info, "Restoring DHT Instance from snapshot")
             catchErrors $
                 createKademliaFromSnapshot bindAddr extAddr kademliaConfig =<<
                 (either error identity . decodeFull) <$> BS.readFile dumpFile
         Nothing -> do
-            logInfo "Creating new DHT instance"
+            traceWith logTrace (Info, "Creating new DHT instance")
             catchErrors $ createKademlia bindAddr extAddr kdiKey kademliaConfig
 
-    logInfo "Created DHT instance"
+    traceWith logTrace (Info, "Created DHT instance")
     let kdiInitialPeers = kpPeers
     let kdiExplicitInitial = kpExplicitInitial
     kdiKnownPeersCache <- atomically $ newTVar []
     pure $ KademliaDHTInstance {..}
   where
     catchErrorsHandler e = do
-        logError $ sformat ("Error launching kademlia with options: "%shown%": "%shown) kconf e
-        throwM e
-    catchErrors x = liftIO x `catchAny` catchErrorsHandler
+        traceWith logTrace (Error, sformat ("Error launching kademlia with options: "%shown%": "%shown) kconf e)
+        throwIO e
+    catchErrors x = x `catchAny` catchErrorsHandler
```

Comment:
Is there a way to refactor that so that its more obvious that its catching `SomeException`, logging it and then re-throwing? Maybe by combining `catchErrors` and `catchErrorsHandler` into a single function?

Functions that catch `SomeException` always make me a bit nervous and I take a much closer look at the code.

Reply:
I'll just replace it with `onException`

Reply:
Oh right, `onException` doesn't get the exception... I'll figure something out.